### ISSUE TITLE
Remove co-location requirement for transaction segments

### ISSUE THREAD

**Issue by tkaitchuck** [pravega/pravega#1049]
**Problem description**
Today transaction segments must be routed to the same container as their parent segment. This means the code needs to know the segment they may be merged into at the time of their creation, and that they cannot be distributed across the cluster.

If this requirement were lifted it could open up more features on the client potentially including:
* Nested transactions
* Transactions that merge into multiple parents
* Scaling operations happening while transactions are in progress
* Better load distribution
* A txn per-writer model where writers all merge into one master segment.

**Suggestions for an improvement**
We could eliminate this requirement by breaking up the merging process into two phases:
Flush and Merge. Once the segment is sealed, flush can be invoked for force all of its data to be written to tier-2. Once this is done, the data is visible to any server, and is immutable. So any process can merge the segment into any other segment.
